core/src/main/scala/scala/pickling/internal/package.scala

```diff
@@ -24,10 +23,18 @@ package object internal {
 
     }
   }
-  private[this] var currentRuntimeVar = new AtomicReference[spi.PicklingRuntime](initDefaultRuntime)
-  def currentRuntime: spi.PicklingRuntime = currentRuntimeVar.get
-  // Here we inject a new runtime for usage.
-  def replaceRuntime(r: spi.PicklingRuntime): Unit = {
+  private[this] var currentRuntimeVar = new AtomicReference[PicklingRuntime](initDefaultRuntime)
+  def currentRuntime: PicklingRuntime = currentRuntimeVar.get
+
+  /** Replace the old [[PicklingRuntime]] keeping its state.
+    *
+    * Note that we don't do anything with the [[RefRegistry]] because in
+    * future versions we are going to change how cyclic references work.
+    */
+  def replaceRuntime(r: PicklingRuntime): Unit = {
+    if(r.picklers.isLookupEnabled) {
+      currentRuntime.picklers.dumpStateTo(r.picklers)
```

Comment:
I think that this is good for a short-term solution, but we should think about a new "cake" approach, because the way to customize the runtime is not sleek. Especially if this runtime strategy depends on implicits that the user may put into scope (like `StaticOnly`, I'll open an issue to explain what I mean by this).
  
But, atm, I agree that we should stick with this and greatly document it. I do not see any other alternative solution.